libs/pilight/events/actions/switch.c

```diff
@@ -174,9 +174,9 @@ static int checkArguments(struct rules_actions_t *obj) {
 								break;
 							}
 						}
-						array_free(&array, l);
 						if(match == 0) {
 							logprintf(LOG_ERR, "switch action \"%s\" is not a valid unit", array[1]);
+							array_free(&array, l);
 							return -1;
 						}
```

Comment:
It should be on line 181 as well or we get a memory leak when there are no errors.